### PR TITLE
feat(AU-1894): send track event when requesting translation

### PR DESCRIPTION
**Pull request by Rodra**
https://2u-internal.atlassian.net/browse/AU-1894

### CODE REVIEW

File: plugins/UnitTranslationPlugin/translation-selection/index.jsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React, { useContext, useEffect } from 'react';
 import PropTypes from 'prop-types';
 
+import { sendTrackEvent } from '@edx/frontend-platform/analytics';
```
Comment (BrandonHBodine) [inline plugins/UnitTranslationPlugin/translation-selection/index.jsx:4]: What's the destination of this? Does it go to a specific platform?

Comment (Rodra) [reply plugins/UnitTranslationPlugin/translation-selection/index.jsx:4]: My understanding is that `sendTrackEvent` will send the event to Segment. These [docs](https://openedx.atlassian.net/wiki/spaces/AN/pages/13205895/Event+Design+and+Review+Process) may be out of date, but it’s what I’m using to differentiate the two. We use `sendTrackEvent` in the Open Courses Xpert Learning Assistant, for what it’s worth, and that’s logged to Segment and then made available in Snowflake.

File: plugins/UnitTranslationPlugin/translation-selection/index.jsx [hunk 2]
```diff
@@ -30,6 +31,20 @@ const TranslationSelection = ({
     language,
   });
 
+  useEffect(() => {
+    if (courseId && language && selectedLanguage && unitId && userId) {
+      const eventName = 'edx.whole_course_translations.translation_requested';
+      const eventProperties = {
```
Comment (leangseu-edx) [inline plugins/UnitTranslationPlugin/translation-selection/index.jsx:37]: I am not sure if we should just send the event when `sourceLanguage !== targetLanguage`. I am pretty sure we can identify all of the users that didn't use the feature with page view.

File: plugins/UnitTranslationPlugin/translation-selection/index.jsx [hunk 3]
```diff
@@ -30,6 +31,20 @@ const TranslationSelection = ({
...
+        courseId,
+        sourceLanguage: language,
+        targetLanguage: selectedLanguage,
+        unitId,
+        userId,
```
Comment (leangseu-edx) [inline plugins/UnitTranslationPlugin/translation-selection/index.jsx:42]: Remove `userId` it is PII.

File: plugins/UnitTranslationPlugin/translation-selection/index.jsx (hunk 2 above)
Comment (Rodra) [reply plugins/UnitTranslationPlugin/translation-selection/index.jsx:37]: Yes I think adding that condition is OK, didn't understand last sentence. What's page view ?

File: plugins/UnitTranslationPlugin/translation-selection/index.jsx (hunk 3 above)
Comment (Rodra) [reply plugins/UnitTranslationPlugin/translation-selection/index.jsx:42]: I had it as needed in the AC, could we still calculate engagement/completion accurately later ?

**Review by leangseu-edx — APPROVED** [review]

**Pull request merged**